src/includes/class-alma-wc-plugin.php

```diff
@@ -761,4 +766,115 @@ public function get_cart_eligibilities() {
 	public function is_cart_eligible() {
 		return count( $this->get_eligible_plans_keys_for_cart() ) > 0;
 	}
+
+
+	/**
+	 * Filter the alma gateway title (visible on checkout page).
+	 *
+	 * @param string  $title The original title.
+	 * @param integer $id The payment gateway id.
+	 * @return string
+	 */
+	public function woocommerce_gateway_title( $title, $id ) {
+
+		if ( 'alma' !== substr( $id, 0, 4 ) ) {
+			return $title;
+		}
+
+		$alma_settings = new Alma_WC_Settings();
+
+		if ( 'alma' === $id ) {
+			$title = $alma_settings->__get( 'title' );
+			if ( ! $title ) {
+				$title = Alma_WC_Settings::default_settings()['title'];
+			}
+		}
+
+		if ( 'alma_pay_later' === $id ) {
+			$title = $alma_settings->__get( 'title_alma_pay_later' );
+			if ( ! $title ) {
+				$title = Alma_WC_Settings::default_settings()['title_alma_pay_later'];
+			}
+		}
+
+		if ( 'alma_more_than_four_instalments' === $id ) {
+			$title = $alma_settings->__get( 'title_alma_more_than_four_instalments' );
+			if ( ! $title ) {
+				$title = Alma_WC_Settings::default_settings()['title_alma_more_than_four_instalments'];
+			}
+		}
+
+		return $title;
+	}
+
+
+	/**
+	 * Filter the alma gateway description (visible on checkout page).
+	 *
+	 * @author Gilles Dumas
+	 * @param  string  $description The original description.
+	 * @param  integer $id The payment gateway id.
+	 * @return string
+	 */
+	public function woocommerce_gateway_description( $description, $id ) {
+
+		if ( 'alma' !== substr( $id, 0, 4 ) ) {
+			return $description;
+		}
+
+		$alma_settings = new Alma_WC_Settings();
+
+		if ( 'alma' === $id ) {
+			$description = $alma_settings->__get( 'description' );
+			if ( ! $description ) {
+				$description = Alma_WC_Settings::default_settings()['description'];
+			}
+		}
+
+		if ( 'alma_pay_later' === $id ) {
```

Comment:
add constants instead of manipulate string everywhere

`const ALMA_GATEWAY_PAY_LATER = 'alma_pay_later';`

`const ALMA_GATEWAY_PAY_MORE_THAN_FOUR = 'alma_more_than_four_instalments';`